velox/exec/WindowPartition.cpp

```diff
@@ -136,17 +170,59 @@ std::pair<vector_size_t, vector_size_t> WindowPartition::computePeerBuffers(
     vector_size_t prevPeerStart,
     vector_size_t prevPeerEnd,
     vector_size_t* rawPeerStarts,
-    vector_size_t* rawPeerEnds) const {
-  auto peerCompare = [&](const char* lhs, const char* rhs) -> bool {
-    return compareRowsWithSortKeys(lhs, rhs);
+    vector_size_t* rawPeerEnds) {
+  auto peerCompare =
+      [&](const char* lhs, const char* rhs, RowContainer* data) -> bool {
+    return compareRowsWithSortKeys(lhs, rhs, data);
   };
 
-  VELOX_CHECK_LE(end, numRows());
+  VELOX_CHECK_LE(end, numRows() + startRow());
 
-  auto lastPartitionRow = numRows() - 1;
+  auto lastPartitionRow = numRows() + startRow() - 1;
   auto peerStart = prevPeerStart;
   auto peerEnd = prevPeerEnd;
-  for (auto i = start, j = 0; i < end; i++, j++) {
+
+  auto nextStart = start;
+  if (previousGroupLastRow_) {
+    // rowContainer is used to compare whether the current first row is in same
+    // peer with the last row of previous batch.
+    auto rowContainer =
+        std::make_unique<RowContainer>(data_->keyTypes(), data_->pool());
+    auto serialized = BaseVector::create<FlatVector<StringView>>(
+        VARBINARY(), 1, data_->pool());
+    data_->extractSerializedRows(
```

Comment:
This presumes that the last row from the previous partial partition is still retained in data_. But that could be erased by clearOutputRows. I think you would need to add a new member variable for a temporary RowContainer and store the last row of the partial partition in it.

Reply:
@aditi-pandit Here is to extract the first row in current partial partition. And the last row in previous partial partition is already stored in previousGroupLastRow_ [here](https://github.com/facebookincubator/velox/pull/9025/files#diff-afebf9a7a0d6dbd1b2c2cb98b36af7a32be8e35a92aa04629db5582519e5003fR259).

Reply:
My bad. Thanks.